torch/utils/data/dataloader.py

```diff
@@ -116,14 +116,23 @@ class DataLoader(Generic[T_co]):
                  :ref:`multiprocessing-best-practices` on more details related
                  to multiprocessing in PyTorch.
 
-    .. note:: ``len(dataloader)`` heuristic is based on the length of the sampler used.
-              When :attr:`dataset` is an :class:`~torch.utils.data.IterableDataset`,
-              ``len(dataset)`` (if implemented) is returned instead, regardless
-              of multi-process loading configurations, because PyTorch trust
-              user :attr:`dataset` code in correctly handling multi-process
-              loading to avoid duplicate data. See `Dataset Types`_ for more
-              details on these two types of datasets and how
-              :class:`~torch.utils.data.IterableDataset` interacts with `Multi-process data loading`_.
+    .. warning:: ``len(dataloader)`` heuristic is based on the length of the sampler used.
+                 When :attr:`dataset` is an :class:`~torch.utils.data.IterableDataset`,
+                 it instead reutrns an estimate based on ``len(dataset) / batch_size``, with proper
+                 rounding depending on :attr:`drop_last`, regardless of multi-process loading
+                 configurations. This represents the best guess PyTorch can make because PyTorch
+                 trust user :attr:`dataset` code in correctly handling multi-process
```

Comment:
"trust user dataset code" -> "trusts user dataset code", because "PyTorch" is singular